lib/resources/wmi.rb

```diff
@@ -40,36 +48,69 @@ def method_missing(*keys)
       keys.shift if keys.is_a?(Array) && keys[0] == :[]
 
       # map all symbols to strings
-      keys = keys.map(&:to_s) if keys.is_a?(Array)
+      keys = keys.map { |x| x.to_s.downcase } if keys.is_a?(Array)
 
       value(keys)
     end
 
     def value(key)
-      extract_value(key, info)
+      extract_value(key, params)
     end
 
-    def info
+    def params
       return @content if defined?(@content)
       @content = {}
 
-      # we should abort execution, if wmi class is not given or wmi resource is
-      # executed on a non-windows system
-      return @content if @wmiclass.nil?
+      # abort if no options are available
+      return @content if @options.nil?
```

Comment:
lets move the skip to top in initialize. then it will be nil for all non-supported operating systems